compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -72,33 +69,33 @@ impl<'r, 'a> AccessLevelsVisitor<'r, 'a> {
                 }
             };
 
-        let module_level = self.r.access_levels.map.get(&module_id).copied();
-        assert!(module_level >= Some(AccessLevel::Exported));
+        let module = self.r.get_module(module_id.to_def_id()).unwrap();
+        let resolutions = self.r.resolutions(module);
 
-        if let Some(exports) = self.r.reexport_map.get(&module_id) {
-            let pub_exports = exports
-                .iter()
-                .filter(|ex| ex.vis == Visibility::Public)
-                .cloned()
-                .collect::<Vec<_>>();
+        for (.., name_resolution) in resolutions.borrow().iter() {
+            if let Some(binding) = name_resolution.borrow().binding() {
```

Comment:
The logic in this block can be simplified a lot, to something like
```rust
                if binding.vis.is_public() && !binding.is_ambiguity() {
                    set_import_binding_access_level(self, binding, module_level);
                    if let Some(def_id) = binding.res().opt_def_id().and_then(|id| id.as_local()) {
                        self.set_access_level_def_id(
                            def_id,
                            if binding.is_import() {
                                Some(AccessLevel::Exported)
                            } else {
                                module_level
                            },
                        );
                    }
                }
```
or maybe even more if `set_import_binding_access_level` is inlined.